### PR TITLE
Fix resizing bug in OWL-ViT

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
Fixes a resizing issue in `OwlViTFeatureExtractor` that lead to the image/s getting resized along only one dimension and getting cropped along the other dimension later on in the preprocessing pipeline. The issue was due to defining the size as a single value instead of a tuple (768 instead of (768, 768)). 

The configuration files are updated and the `OwlViTProcessor `can correctly resize the input images now. This PR changes the default target `size` and sets default `do_center_crop` argument as False.

Fixes # 18553

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

@NielsRogge could you take a look?

### CODE REVIEW

File: src/transformers/models/owlvit/feature_extraction_owlvit.py [hunk 1]
```diff
@@ -50,13 +50,13 @@ class OwlViTFeatureExtractor(FeatureExtractionMixin, ImageFeatureExtractionMixin
     Args:
         do_resize (`bool`, *optional*, defaults to `True`):
             Whether to resize the shorter edge of the input to a certain `size`.
-        size (`int`, *optional*, defaults to 768):
+        size (`int`, *optional*, defaults to (768, 768):
```
Comment (NielsRogge) [inline src/transformers/models/owlvit/feature_extraction_owlvit.py:53]:
```suggestion
        size (`int`, *optional*, defaults to (768, 768)):
```

File: src/transformers/models/owlvit/feature_extraction_owlvit.py [hunk 2]
```diff
@@ -50,13 +50,13 @@ class OwlViTFeatureExtractor(FeatureExtractionMixin, ImageFeatureExtractionMixin
...
             Resize the shorter edge of the input to the given size. Only has an effect if `do_resize` is set to `True`.
```
Comment (NielsRogge) [inline src/transformers/models/owlvit/feature_extraction_owlvit.py:54]: Would be great to clarify here whether it should be (width, height) or (height, width).

Normally, it should be (width, height), as we're using Pillow for resizing, right?

**Review by NielsRogge — APPROVED** [review]
Thanks for fixing! However, all "owlvit" checkpoints on the hub have `"do_center_crop": true` in their preprocessor_config.json.

Should we update them after this PR is merged?

File: src/transformers/models/owlvit/feature_extraction_owlvit.py (hunk 2 above)
Comment (alaradirik) [reply src/transformers/models/owlvit/feature_extraction_owlvit.py:54]: Yes, exactly! I will clarify it to avoid confusion though

**Comment by alaradirik** [comment]
> Thanks for fixing! However, all "owlvit" checkpoints on the hub have `"do_center_crop": true` in their preprocessor_config.json.
> 
> Should we update them after this PR is merged?

Good point, yes, we should update them after the merge

**Pull request merged**